### PR TITLE
bpo-40899: Document exception raised when module cannot be imported

### PR DESCRIPTION
**Pull request by dukecat0**
https://bugs.python.org/issue40899

### CODE REVIEW

**Review by ambv — CHANGES_REQUESTED** [review]
1. Note Ronald's comment on arbitrary exceptions being raised from a module body while it's being imported.
2. Please don't insert error handling information in the leading paragraph. Look at other functions in functions.rst, they have exception information listed further in the text, usually as the last paragraph describing them.

**Review by gpshead — APPROVED** [review]

**Comment by gpshead** [comment]
> 1. Note Ronald's comment on arbitrary exceptions being raised from a module body while it's being imported.
> 2. Please don't insert error handling information in the leading paragraph. Look at other functions in functions.rst, they have exception information listed further in the text, usually as the last paragraph describing them.

while I've Approved this change from the standpoint of not being a bad thing to document, I do think the above should be addressed first.

**Comment by iritkatriel** [comment]
@meowmeowmeowcat This has been awaiting changes for over a year. Are you planning to finish this or shall we close it so someone else can pick it up?

**Review by iritkatriel — CHANGES_REQUESTED** [review]
See https://github.com/python/cpython/pull/27709#issuecomment-901432942.

**Comment by meowmeowmeowcat** [comment]
> Are you planning to finish this or shall we close it so someone else can pick it up?

I am going to finish this. Sorry for the late response!

I have made the requested changes; please review again

File: Doc/library/importlib.rst [hunk 1]
```diff
@@ -124,6 +124,9 @@ Functions
     need to call :func:`invalidate_caches` in order for the new module to be
     noticed by the import system.
 
+    A :exc:`ImportError` can be raised if the module cannot be imported
```
Comment (dukecat0) [inline Doc/library/importlib.rst]:
```suggestion
    A :exc:`ImportError` exception can be raised if the module cannot be imported
```

File: Doc/library/functions.rst [hunk 2]
```diff
@@ -2018,6 +2018,9 @@ are always available.  They are listed here in alphabetical order.
    object, the names to import are retrieved and assigned to their respective
    names.
 
+   A :exc:`ImportError` exception can be raised if the module cannot be imported
```
Comment (iritkatriel) [inline Doc/library/functions.rst]:
```suggestion
   An :exc:`ImportError` exception can be raised if the module cannot be imported
```

File: Doc/library/importlib.rst [hunk 3]
```diff
@@ -124,6 +124,9 @@ Functions
     need to call :func:`invalidate_caches` in order for the new module to be
     noticed by the import system.
 
+    A :exc:`ImportError` exception can be raised if the module cannot be imported
```
Comment (iritkatriel) [inline Doc/library/importlib.rst]:
```suggestion
    An :exc:`ImportError` exception can be raised if the module cannot be imported
```

File: Doc/library/functions.rst [hunk 4]
```diff
@@ -2018,6 +2018,9 @@ are always available.  They are listed here in alphabetical order.
    object, the names to import are retrieved and assigned to their respective
    names.
 
+   An :exc:`ImportError` exception can be raised if the module cannot be imported
+   successfully.
```
Comment (iritkatriel) [inline Doc/library/functions.rst:2022]: I'm not sure I like this wording - what do you mean "can be raised", it sounds like it is possible for import to fail and not raise an exception.

Comment (dukecat0) [reply Doc/library/functions.rst:2022]: > Note Ronald's comment on arbitrary exceptions being raised from a module body while it's being imported.

It means arbitrary exceptions can also be raised.

Comment (iritkatriel) [reply Doc/library/functions.rst:2022]: I understand the intent, but I don't think that's the only possible interpretation of the current wording.

Comment (dukecat0) [reply Doc/library/functions.rst:2022]: I just found out that https://github.com/python/cpython/pull/94662 is also doing the same thing. Would you prefer that one?

**Pull request closed without merging**